docs/basics/gas.md

```diff
@@ -80,6 +85,19 @@ types other than the ones defined by the `evm` module. EVM module transactions c
 as gas is required inherently by the EVM. This check is done by the EVM transactions stateless validation
 (i.e `ValidateBasic`) function as well as on the custom `AnteHandler` defined by Ethermint.
 
+## Gas estimation
+
+Ethereum provides a JSON-RPC endpoint `eth_estimateGas` to help users set up a correct gas limit in their transactions. 
+
+Unfortunately we cannot make use of the SDK `tx simulation` for gas estimation because the pre-check in the Ante Handlers would require a valid signature, and the sender balance to be enough to pay for the gas. But in Ethereum, this endpoint can be called without specifying any sender address.
+
+For that reason, a specific query api `EstimateGas` is implemented in Ethermint. It will apply the transaction against the current block/state and perform a binary search in order to find the optimal gas value to return to the user (the same transaction will be applied over and over until we find the minimum gas needed before it fails). The reason we need to use a binary search is that the gas requirement for the
```

Comment:
```suggestion
For that reason, a specific query API `EstimateGas` is implemented in Ethermint. It will apply the transaction against the current block/state and perform a binary search in order to find the optimal gas value to return to the user (the same transaction will be applied over and over until we find the minimum gas needed before it fails). The reason we need to use a binary search is that the gas required for the
```